src/backends/gpu/kernels/nernst.cu

```diff
@@ -0,0 +1,36 @@
+#include <cstdint>
+
+#include "../nernst.hpp"
+#include "detail.hpp"
+
+namespace arb {
+namespace gpu {
+
+namespace kernels {
+    template <typename T>
+    __global__ void nernst(std::size_t n, int valency, T temperature, const T* Xo, const T* Xi, T* eX) {
+        auto i = threadIdx.x+blockIdx.x*blockDim.x;
+
+        constexpr T RF = 1e3*8.3144598/96485.33289; // factor 1e3 to scale from V -> mV
+        T factor = RF*temperature/valency;
+        if (i<n) {
+            eX[i] = factor*std::log(Xo[i]/Xi[i]);
+        }
```

Comment:
As we saw in the NEURON code, there the computed potential is clamped (for when one of the concentrations is zero, or (improperly) negative). Should we also?

Reply:
how about just clamping the concentrations, e.g. 
```
constexpr value_type min_conc = 1e-6;
log(max(Xo[i], min_conc)/max(Xi[i], min_conc))
```

Reply:
From private discussion, we have decided to leave it as is, and see how/when the assumptions of the Nernst equation break down, before proposing solutions.